libdevcore/LevelDB.cpp

```diff
@@ -29,6 +29,32 @@ inline leveldb::Slice toLDBSlice(Slice _slice)
     return leveldb::Slice(_slice.data(), _slice.size());
 }
 
+template <class _T>
+void throwDatabaseError(leveldb::Status const& _status, boost::filesystem::path const& _path)
```

Comment:
I don't see reason for this helper function. Do this preprocessing in checkStatus().